### PR TITLE
CALS-5467: Applicant Marital History now stores to database on Save Progress

### PR DESCRIPTION
**Pull request by LeonidMarushevskyi**
#### Which User story does this relate to?
- https://osi-cwds.atlassian.net/browse/CALS-5467

#### Brief feature description
-RFA 1A VI. Marital History card fixed to save data to the database and retain the data after Save Progress click. In addition, several other cards fixed to retain the data after Save Progress: IV. Minor Children and VIII. Foster Care / Adoption / Licensure History. Rails A01 API modified to match with backend /rfa-1a-forms API

#### Does this PR block/effect other stories?
-No

#### Do any commands need to be run in order to run the project with this PR?
-No

#### Code coverage percent diff as a result of this PR?
- No

### CODE REVIEW

**Comment by LeonidMarushevskyi** [comment]
Nevermind this pull requests for now. Working on tests fixes.

**Comment by LeonidMarushevskyi** [comment]
Working on Code Climate issues

**Comment by Nasir2712** [comment]
As per our discussion, updating the rfa01a controller method should fix the most of the issues.

**Comment by LeonidMarushevskyi** [comment]
@Nasir2712 I have applyed suggested change to controller to decouple backend API from Rails API.

**Comment by LeonidMarushevskyi** [comment]
Please squash commits when will merge the PR.

File: app/controllers/rfa/a01_controller.rb [hunk 1]
```diff
@@ -77,22 +77,53 @@ def residence_params
...
+      next unless 0.eql?(former_spouse['applicant_id'].to_i)
+        former_spouse.merge!(ActionController::Parameters
+            .new('applicant_id' => applicant_names
+                .key(former_spouse['applicant_id'])).permit!)
+    end
+    applicants_history
+  end
+
+  def set_applicant_id_for_adult_children(applicant_names, applicants_history)
+    applicants_history['adult_children'].each do |adult_child|
```
[the remark this answers is not in the document]
Comment (LeonidMarushevskyi) [reply app/controllers/rfa/a01_controller.rb]: @ratneshraval Code changed to reused set_relationship_to_applicants for adult_children portion of Marital history which is similar to all other cards. For former_spouses portion, stil need a different way to iterate over so added specific method in a01_controller.

**Comment by ratneshraval** [comment]
@LeonidMarushevskyi the method `set_relationship_to_applicants` can be refactored to use for all 3 relationships. You'll need to add key name to the method so it is not tied to `relationship_to_applicants'][0]` the key can supply value of `relationship_to_applicants'][0]` or `former_spouses`

**Comment by Nasir2712** [comment]
@LeonidMarushevskyi I see you removed the rfa01 controller updated method. When i tested the application everything was working with that just one change. Did you try that approach?

File: app/controllers/cals_base_controller.rb [hunk 2]
```diff
@@ -33,12 +33,12 @@ def create_update_or_delete(item, helper, parent_id)
     end
   end
 
-  def set_relationship_to_applicants(parameters, response)
-    applicant_names = response.map { |app| [app.id, "#{app.first_name} #{app.middle_name} #{app.last_name}".squish] }.to_h
-    if 0.eql?(parameters['relationship_to_applicants'][0]['applicant_id'].to_i)
-      parameters['relationship_to_applicants'][0].merge!(ActionController::Parameters.new('applicant_id' => applicant_names.key(parameters['relationship_to_applicants'][0]['applicant_id'])).permit!)
+  def set_relationship_to_applicants(object, applicants)
```
Comment (lauherkcambria) [inline app/controllers/cals_base_controller.rb]: please use another naming convention than 'object'

File: app/controllers/rfa/a01_controller.rb [hunk 3]
```diff
@@ -33,11 +33,11 @@ def update
     @application_response[:application_county] = process_items_for_persistance(application_county_params, rfa_application_helper, params[:id]) if params[:application_county].present?
     @application_response[:applicants] = process_items_for_persistance(applicant_params, rfa_applicant_helper, params[:id]) if params[:applicants].present?
     @application_response[:residence] = process_items_for_persistance(residence_params, rfa_residence_helper, params[:id]) if params[:residence].present?
-    @application_response[:applicantsHistory] = process_items_for_persistance(applicants_history_params, rfa_applicant_history_helper, params[:id]) if params[:applicantsHistory].present?
-    @application_response[:minorChildren] = process_items_for_persistance(minor_children_params, rfa_minor_children_helper, params[:id]) if params[:minorChildren].present?
+    @application_response[:applicants_history] = process_items_for_persistance(applicants_history_params, rfa_applicant_history_helper, params[:id]) if params[:applicants_history].present?
```
Comment (lauherkcambria) [inline app/controllers/rfa/a01_controller.rb]: thanks for addressing these casings.

File: app/controllers/cals_base_controller.rb (hunk 2 above)
Comment (ratneshraval) [reply app/controllers/cals_base_controller.rb]: I agree. let's use something more descriptive.

Comment (LeonidMarushevskyi) [reply app/controllers/cals_base_controller.rb]: Naming convention fixed. parameters name returned back

**Pull request merged**